base/sequence_clock.go

```diff
@@ -291,6 +292,23 @@ func (c *SequenceClockImpl) UpdateWithClock(updateClock SequenceClock) {
 	}
 }
 
+func (c *SequenceClockImpl) UpdateWithPartitionClocks(partitionClocks []*PartitionClock, allowRollback bool) {
+
+	for _, partitionClock := range partitionClocks {
+		if partitionClock != nil {
+			for vbNo, seq := range *partitionClock {
+				if seq > 0 {
+					if seq < c.value[vbNo] && !allowRollback {
+						panic(fmt.Sprintf("Update attempted to set clock to earlier sequence.  Vb: %d, currentSequence: %d, newSequence: %d", vbNo, c.value[vbNo], seq))
```

Comment:
Yeah I think the panic is fine, was just thinking we could push it up higher in the stack at some point.